evalml/pipelines/components/transformers/preprocessing/stl_decomposer.py

```diff
@@ -442,6 +442,7 @@ def inverse_transform(
             y.append(y_series)
         y_df = pd.DataFrame(y).T
         y_df.index = original_index
+        y_df.columns = y_t.columns
```

Comment:
The predictions weren't getting the corresponding series ID values as the column names and that's needed since the decomposer uses this to select the correct value. Before this was causing the decomposer to return NaN values. @christopherbunn figured that out so he might have more info.

Reply:
The predictions that are generated do not have the series ID values as their column names. Copying these names over is required so we can inverse_transform from the decomposer.